src/battle_ai_main.c

```diff
@@ -3451,7 +3459,7 @@ static u32 AI_CalcMoveEffectScore(u32 battlerAtk, u32 battlerDef, u32 move)
         }
         break;
     case EFFECT_REST:
-        if (!(CanBeSlept(battlerAtk, aiData->abilities[battlerAtk])))
+        if (!(CanBeSleptOrDrowsy(battlerAtk, aiData->abilities[battlerAtk])))
```

Comment:
We should probably clean up this line while we're at it and remove a pair of parentheses